### PR TITLE
20 create student page

### LINKED ISSUE #20: Create student page

### LINKED ISSUE #21: Create student settings page

### PR DESCRIPTION
**Pull request by SukhanovPavel**
fix: 
add css modules
add pattern for tg nickname input
add store for user and user settings
delete 'onChange' in input
delete style prop, delete comments
add ESLint

### CODE REVIEW

File: src/pages/student/page.tsx [hunk 1]
```diff
@@ -1,11 +1,74 @@
...
+            </Row>
+          </Col>
+          <Space size={10} className={styles.links} direction="vertical">
+            <Link className={styles.button} to={routes.studentSettings} >
+              <Text><SettingOutlined  className={styles.icon}/></Text>
+            </Link>
+            <Link to={`https://t.me/${userSettings.tgNickName.slice(1)}`} ><u>tg: {userSettings.tgNickName || 'nickname'}</u></Link>
+            <Link to={`https://github.com/${userSettings.githubNickName}`} ><u>github: {userSettings.githubNickName || 'nickname'}</u></Link>
+            <Link to={routes.studentSettings} ><u>{userSettings.htmlRepository || 'repository html'}</u></Link>
+            <Link to={routes.studentSettings} ><u>{userSettings.reactRepository || 'repository react'}</u></Link>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx]: Link to={`https://github.com/${userSettings.githubNickName}/${userSettings.htmlRepository}`}
Link to={`https://github.com/${userSettings.githubNickName}/${userSettings.reactRepository}`}

File: src/pages/student/page.tsx [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
+  ]);
+
+  return (
   <Layout>
     <PageHeader title="Личный кабинет" />
-    <Content />
+    <Content className={styles._}>
+      <Row>
+        <Space size={100}>
+          <Col span={50}>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx]: span max 24, maybe it's better to use one col for table and the second for user links?

File: src/pages/student/page.tsx [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
+                <Card.Grid key={card.deadline} className={styles.gridStyle}>
+                  <Text
+                    // type={card.homeworkStatus}
+                  >
+                    {card.homework}
+                  </Text>
+                </Card.Grid>
+              ))}
+            </Card>
+            <Row gutter={20}>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx]: Table beside Card.Grid?

File: src/pages/student/model.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  tgNickName: '',
+  githubNickName: '',
+  htmlRepository: '',
+  reactRepository: ''
+  }
+);
+
+export const setUserSettings = createEvent<UserSettings>();
+
+$userSettings.on(setUserSettings, ( _, payload ) => payload)
```
Comment (Hellomyguest) [inline src/pages/student/model.ts]: maybe this way is better:
image

File: src/pages/student/settings/page.tsx [hunk 5]
```diff
@@ -1,11 +1,137 @@
...
+
+  return(
   <Layout>
     <PageHeader title="Настройки" />
-    <Content />
+    <Content className={styles._}>
+      <Form
+        name="userSettings"
+        className="Form"
+        initialValues={{ remember: true }}
```
Comment (Hellomyguest) [inline src/pages/student/settings/page.tsx]: initialValues={userSettings} and then delete all the input values props

File: src/pages/student/page.module.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+.title {
+  margin-top: 0;
+}
+
+.icon {
+  font-size: 3rem;
+}
+
+.cardBody {
+  width: 50rem;
```
Comment (Hellomyguest) [inline src/pages/student/page.module.css:13]: why?
screen size 1024px -  settings broke

File: src/pages/student/page.tsx [hunk 7]
```diff
@@ -1,11 +1,83 @@
...
+                    >
+                      {card.homework}
+                    </Text>
+                  </Card.Grid>
+                ))}
+              </Card>
+              <Row gutter={20}>
+                {homeworkCards.map((card) => (
+                  <Col key={card.deadline} className={styles.gridStyle}>
+                    <div>{card.homeworkStatus ? "" : card.deadline}</div>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx]: why div? won't it be better to use one homeworksCards.map => card with content in grid system?
[https://codesandbox.io/s/7r6080?file=/demo.tsx:249-261](url)

File: src/pages/student/settings/page.tsx [hunk 8]
```diff
@@ -1,11 +1,105 @@
...
+            ]}
+          >
+            <Input placeholder="ник в телеграме" allowClear />
+          </Form.Item>
+          <Form.Item
+            name="htmlRepository"
+            rules={[
+              {
+                required: true,
+                message: "Please input name of repository!",
```
Comment (Hellomyguest) [inline src/pages/student/settings/page.tsx]: if i just register and don't have repositories yet - i can't fill this.

File: src/pages/student/settings/page.tsx [hunk 9]
```diff
@@ -1,11 +1,105 @@
...
+            ]}
+          >
+            <Input placeholder="имя репозитория html" allowClear />
+          </Form.Item>
+          <Form.Item
+            name="reactRepository"
+            rules={[
+              {
+                required: true,
+                message: "Please input name of repository!",
```
Comment (Hellomyguest) [inline src/pages/student/settings/page.tsx]: if i just register and don't have repositories yet - i can't fill this.

File: src/pages/student/page.tsx [hunk 10]
```diff
@@ -1,11 +1,83 @@
...
+        <Row>
+          <Space size={100}>
+            <Col span={24}>
+              <Title className={styles.title}>
+                Привет, {user.name} {user.surname} ({user.nickName})
+              </Title>
+              <Title level={3}>Оценки за дз</Title>
+              <Card className={styles.cardBody}>
+                {homeworkCards.map((card) => (
+                  <Card.Grid key={card.deadline} className={styles.gridStyle}>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx:30]: to understand that homework is acceptet u can do it like this: 
<Card.Grid key={card.deadline} className={`${styles.gridStyle} ${card.homeworkStatus ? styles.green : null}`}>

but in our case we also have yellow one, so make enum.
also if we need 2 or more classes i think it's better to use classnames plugin

File: src/pages/student/page.tsx [hunk 11]
```diff
@@ -1,11 +1,83 @@
...
+                    <Text
+                    // type={card.homeworkStatus}
+                    >
+                      {card.homework}
+                    </Text>
```
Comment (Hellomyguest) [inline src/pages/student/page.tsx]: to understand that homework is accepted u can do it like this:
image

but in our case we also have yellow one, so make enum.

File: src/pages/student/model.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    htmlRepository: "",
+    reactRepository: "",
+  },
+});
+
+export const setUserSettings = createEvent<UserSettings>();
+
+$user.on(setUserSettings, (state, payload) => {
+  const user = { ...state };
+  user.settings = payload;
```
Comment (dpolevodin) [inline src/pages/student/model.ts]: можно попробовать заменить более короткой записью
```
$user.on(setUserSettings, (state, payload) => ({ ...state, settings: payload }))
```

File: src/pages/student/settings/mocks.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
+export type HomeworkCards = {
```
Comment (dpolevodin) [inline src/pages/student/settings/mocks.ts]: Если тип для одной карточки лучше использовать название```HomeworkCard```

File: src/pages/student/settings/page.tsx [hunk 14]
```diff
@@ -1,11 +1,105 @@
...
-  </Layout>
-);
+export const StudentSettingsPage = () => {
+  const [user, setUserSettingsFn, toStudentPageFn] = useUnit([
+    $user,
+    setUserSettings,
+    toStudentPage,
+  ]);
+  const handleFinish = (value: UserSettings) => {
+    setUserSettingsFn(value);
```
Comment (dpolevodin) [inline src/pages/student/settings/page.tsx]: Предлагаю попробовать завязаться на ```setUserSettingsFn``` в логике эффектора для редиректа
Мне кажется, можно избежать дополнительного вызова функции ```toStudentPageFn``` в данном хэндлере

**Review by dpolevodin — APPROVED** [review]

**Pull request merged**